shared/src/balancer/swap/fixed_point.rs

```diff
@@ -36,6 +36,12 @@ impl From<usize> for Bfp {
     }
 }
 
+impl From<Bfp> for usize {
+    fn from(num: Bfp) -> Self {
+        num.as_uint256().checked_div(*ONE_18).unwrap().as_usize()
+    }
+}
```

Comment:
This conversion feels a bit wrong, as you will lose all fractional bits from the fixed point number.

Reply:
I would love to do this, but I don't have any ideas in mind for converting it any differently than I have here....

Reply:
The issue with the code here is that a `Bfp` of `0.5` would become a `BigDecimal` of value `0.0` as you are truncating the fractional bits by dividing by `1.0` (correct me if I'm wrong @fedgiac).

Reply:
Yep, this would truncate any decimal. Why is `usize` from `Bfp` needed anyway, considering that all weights are between 0 and 1? (Now I read the PR.)
The BigDecimal could be computed as numerator: `num.as_uint256()` and denominator: `*ONE_18`.

Reply:
Yea. I'll do this right MEOW!